### PR TITLE
feat: allow user to gets list of his tenants

### PR DESCRIPTION
**Pull request by anatolychernov**
PR by issue [328](https://github.com/projectcapsule/capsule-proxy/issues/328)

### CODE REVIEW

**Comment by anatolychernov** [comment]
It seems I've fixed commit's checks

File: internal/modules/tenants/get.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	"github.com/projectcapsule/capsule-proxy/internal/modules/errors"
+	"github.com/projectcapsule/capsule-proxy/internal/request"
+	"github.com/projectcapsule/capsule-proxy/internal/tenant"
+)
+
+type get struct {
+	capsuleLabel string
+	client       client.Reader
+	log          logr.Logger
+	rbReflector  *controllers.RoleBindingReflector
```
Comment (prometherion) [inline internal/modules/tenants/get.go:30]: `rbReflector` is not required and never used, should be removed.

File: internal/modules/tenants/list.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+	"github.com/projectcapsule/capsule-proxy/internal/controllers"
+	"github.com/projectcapsule/capsule-proxy/internal/modules"
+	"github.com/projectcapsule/capsule-proxy/internal/modules/errors"
+	"github.com/projectcapsule/capsule-proxy/internal/request"
+	"github.com/projectcapsule/capsule-proxy/internal/tenant"
+)
+
+type list struct {
+	roleBindingsReflector *controllers.RoleBindingReflector
```
Comment (prometherion) [inline internal/modules/tenants/list.go:24]: Same here, `roleBindingsReflector` is not required.

**Review by prometherion — CHANGES_REQUESTED** [review]

File: internal/modules/tenants/get.go (hunk 1 above)
Comment (anatolychernov) [reply internal/modules/tenants/get.go]: Fixed

File: internal/modules/tenants/list.go (hunk 2 above)
Comment (anatolychernov) [reply internal/modules/tenants/list.go]: Fixed

**Comment by anatolychernov** [comment]
Hello,

Could you start other checks from workflow?

**Comment by prometherion** [comment]
@anatolychernov `golangci-lint` is complaining since you're relying on the `slices` package which has been introduced with the v1.21.

We should first get the proxy update to this version of Go, then we can get this merged. Are you up to take care of this too?

**Comment by anatolychernov** [comment]
> @anatolychernov `golangci-lint` is complaining since you're relying on the `slices` package which has been introduced with the v1.21.
> 
> We should first get the proxy update to this version of Go, then we can get this merged. Are you up to take care of this too?

Hello,

I've updated Go version to 1.21 in go.mod

**Review by prometherion — CHANGES_REQUESTED** [review]
All the commits have the same subject, making hard to keep track of the changes.

Furthermore, Go 1.21 should be configured also on the CI actions, not only on the `go.mod` file.

e.g.: https://github.com/projectcapsule/capsule/blob/318d7d076a4a7640f6459ffc14d8e965380d344d/.github/workflows/e2e.yml#L46-L48

**Comment by anatolychernov** [comment]
> All the commits have the same subject, making hard to keep track of the changes.
> 
> Furthermore, Go 1.21 should be configured also on the CI actions, not only on the `go.mod` file.
> 
> e.g.: https://github.com/projectcapsule/capsule/blob/318d7d076a4a7640f6459ffc14d8e965380d344d/.github/workflows/e2e.yml#L46-L48

This link from capsule project, not from capsule-proxy. But I've updated go version in Docker  files.

**Comment by anatolychernov** [comment]
@prometherion 

If we don't have any more questions, can we proceed checks?

**Comment by prometherion** [comment]
> This link is from capsule project, not from capsule-proxy

The same Go runtime settings must be reported here: https://github.com/projectcapsule/capsule-proxy/blob/23332530f35f5700312734fb4e94ecee75bb4a34/.github/workflows/lint.yaml#L13-L24

Otherwise, the `golangci-lint` won't be able to access the `slices` package from the standard library.

**Comment by anatolychernov** [comment]
> > This link is from capsule project, not from capsule-proxy
> 
> The same Go runtime settings must be reported here:
> 
> https://github.com/projectcapsule/capsule-proxy/blob/23332530f35f5700312734fb4e94ecee75bb4a34/.github/workflows/lint.yaml#L13-L24
> 
> Otherwise, the `golangci-lint` won't be able to access the `slices` package from the standard library.

Hello @prometherion,

Do you mean something like this?

```golang
jobs:
  golangci:
    name: lint
    runs-on: ubuntu-20.04
    steps:
      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
      - name: Install Go
        uses: actions/setup-go@v4
        with:
          go-version: '1.21'
          cache: false
      - name: Run golangci-lint
        uses: golangci/golangci-lint-action@3a919529898de77ec3da873e3063ca4b10e7f5cc # v3.7.0
        with:
          version: v1.51.2
          only-new-issues: false
          args: --timeout 5m --config .golangci.yml
```

**Comment by anatolychernov** [comment]
@prometherion,

I've added "Install Go" step for lint.yaml

**Comment by anatolychernov** [comment]
@prometherion, could you proceed checks to get 'Go lint' result?

**Comment by anatolychernov** [comment]
Hello guys!
@prometherion,
I have rebased my changes from fresh main commits. Could you proceed other workflow checks?

**Comment by prometherion** [comment]
@anatolychernov do you mind if you `git push --force` some fixes to get this merged?

**Comment by anatolychernov** [comment]
@prometherion,
Do you mean that you want to combine these commits into one(squash) or I didn't understand you. Could you clarify it?

**Comment by prometherion** [comment]
`Maintainers are allowed to edit this pull request`

I can push yo your branch to fix minor thingies, that would be much easier to get this merged, considering that everytime you have to ask for CI to get triggered since you're a first-time contributor 😅

**Comment by anatolychernov** [comment]
> `Maintainers are allowed to edit this pull request`
> 
> I can push yo your branch to fix minor thingies, that would be much easier to get this merged, considering that everytime you have to ask for CI to get triggered since you're a first-time contributor 😅

Yes, please do it. It will be great.

**Comment by anatolychernov** [comment]
> > `Maintainers are allowed to edit this pull request`
> > I can push yo your branch to fix minor thingies, that would be much easier to get this merged, considering that everytime you have to ask for CI to get triggered since you're a first-time contributor 😅
> 
> Yes, please do it. It will be great.

@prometherion hello!
Do you have any updates about minor fixes?

**Comment by anatolychernov** [comment]
I've signed off all my commits, but merging check is still blocked. Is it a glitch?

**Comment by prometherion** [comment]
@anatolychernov I'm sorry for this tedious PR but I think we're there, eventually!

Since you're a first-time contributor, GI must approved by a project administrator, sorry for this.
Also, the latest `golangci-lint` seems having issues with the `slices` packages you used in your contribution: I changed it according to the current libraries we're using in `capsule-proxy` with the same outcome.
Furthermore, I spotted a bug in the `Makefile` which was still using the old `golangci-lint` binary.

Let's praise for the green CI 🙏🏻  and we can finally get this merged: thanks a lot for your patience and your efforts, I appreciate it!

**Review by prometherion — APPROVED** [review]

**Comment by anatolychernov** [comment]
@prometherion Thank you for help!

**Pull request merged**